User login

james-cape opened the pull request:
## What functionality does this accomplish?
Closes #'s:
[7](https://github.com/james-cape/express_sweater_weather/issues/7)
[20](https://github.com/james-cape/express_sweater_weather/issues/20)

**User Login:**
A POST request to `/api/v1/sessions` containing user email and password will return the user's existing API key. 
Sad paths accounted for are missing field entries and mismatching passwords.
@bplantico Can you tell me how I might test for a user that doesn't exist in the database?


## What did you struggle on to complete?
I struggled to instantiate an existing user for test purposes. 
For testing:
```
        let params = {
          "email": "my_email@example.com",
          "password": "password",
        }

        return User.create({
            email: "my_email@example.com",
            password: bcrypt.hashSync("password", 10),
            apiKey: "12345"
        })
        .then (user => {
          return request(app).post("/api/v1/sessions").send(params)
        })
```

For routing:
```
  if (req.body.email && req.body.password) {
      User.findOne({
        where: {
          email: req.body.email
        }
      })
      .then(user => {
        if (bcrypt.compareSync(req.body.password, user.password)) {
          res.setHeader("Content-Type", "application/json");
          res.status(200).send(JSON.stringify({api_key: user.apiKey}));
        } else {
          res.setHeader("Content-Type", "application/json");
          res.status(401).send(JSON.stringify({error: "Incorrect password"}));
        }
      })
etc
```


## Current Test Suite:
### Test Coverage Percentage: x%
- [ ] No Tests have been changed
- [ ] Some Tests have been changed
- [ ] All of the Tests have been changed(Please describe):

## Checklist:
- [ ] My code has no unused/commented out code
- [ ] I have reviewed my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have fully tested my code
- [ ] I have partially tested my code (please explain):

## Helpful Resources:
* Resource link AND small description of info gathered/learned

bplantico commented:
@james-cape If you leave out the part where you create a user in your test and I think your code already accounts for the case!

james-cape commented:
@bplantico Good catch. I went ahead and added a test where I modified the user email so it would not match the on in the POST request params. I also added a more descriptive error message in the case that the email address is not on file. Thanks!
